src/sockperf.cpp

```diff
@@ -3332,26 +3332,42 @@ int bringup(const int *p_daemonize) {
     /* Setup VMA */
     int _vma_pkts_desc_size = 0;
 
-#ifdef USING_VMA_EXTRA_API
+#if defined(USING_VMA_EXTRA_API) || defined(USING_XLIO_EXTRA_API)
     if (!rc && (s_user_params.is_rxfiltercb || s_user_params.is_zcopyread ||
                 s_user_params.fd_handler_type == SOCKETXTREME)) {
         // Get VMA extended API
         g_vma_api = vma_get_api();
-        if (g_vma_api == NULL) {
-            errno = EPERM;
-            exit_with_err("VMA Extra API is not available", SOCKPERF_ERR_FATAL);
+        if (!g_vma_api) {
+            if (s_user_params.is_rxfiltercb || s_user_params.fd_handler_type == SOCKETXTREME) {
+                exit_with_err("VMA Extra API is not available", SOCKPERF_ERR_FATAL);
+            }
+
+            g_xlio_api = xlio_get_api();
+            if (!g_xlio_api) {
```

Comment:
I am not sure that this logic is correct for VMA/XLIO mixed mode that is announced as supported

Reply:
It works. We fetch xlio API only if VMA api couldn't be found.

Reply:
Done

Reply:
looks the same w/o fix

Reply:
Refactored for two seperate Ifs